### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by samdark**
- Add types
- Add strict types
- Fix leftovers in benchmarks
- Remove unused imports
- Remove unused phpdoc
- Correct phpdoc mistakes
- Correct spelling mistakes

| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Tests pass?   | ✔️
| Fixed issues  | -

### CODE REVIEW

File: src/Container.php [hunk 1]
```diff
@@ -22,28 +24,29 @@ final class Container extends AbstractContainerConfigurator implements Container
...
 
     private ?ContainerInterface $rootContainer = null;
 
     /**
      * Container constructor.
      *
-     * @param array $definitions
-     * @param ServiceProviderInterface[] $providers
+     * @param array $definitions Definitions to put into container.
+     * @param ServiceProviderInterface[]|array $providers Service providers to get definitions from.
```
Comment (roxblnfk) [inline src/Container.php:45]:
```suggestion
     * @param ServiceProviderInterface[] $providers Service providers to get definitions from.
```

File: src/Container.php [hunk 2]
```diff
@@ -22,28 +24,29 @@ final class Container extends AbstractContainerConfigurator implements Container
...
      * to detect circular references
      */
-    private $building = [];
+    private array $building = [];
 
     /**
      * @var object[]
      */
-    private $instances;
+    private array $instances;
```
Comment (roxblnfk) [inline src/Container.php:37]:
```suggestion
    private array $instances = [];
```

File: src/Container.php (hunk 1 above)
Comment (samdark) [reply src/Container.php:45]: There is a concept of deferred providers and these are specified as class names. See https://github.com/yiisoft/di/blob/master/tests/Unit/DeferredServiceProviderTest.php#L28

Comment (roxblnfk) [reply src/Container.php:45]: hm... `ServiceProviderInterface[]|string[]` ?

Comment (samdark) [reply src/Container.php:45]: Yes. I'll change it. Thanks.

**Pull request merged**